src/nvim/fileio.c

```diff
@@ -7129,32 +7129,30 @@ int au_exists(char_u *arg)
   return retval;
 }
 
-
-
-/*
- * Try matching a filename with a "pattern" ("prog" is NULL), or use the
- * precompiled regprog "prog" ("pattern" is NULL).  That avoids calling
- * vim_regcomp() often.
- * Used for autocommands and 'wildignore'.
- * Returns TRUE if there is a match, FALSE otherwise.
- */
-int 
-match_file_pat (
-    char_u *pattern,                   /* pattern to match with */
-    regprog_T *prog,                      /* pre-compiled regprog or NULL */
-    char_u *fname,                     /* full path of file name */
-    char_u *sfname,                    /* short file name or NULL */
-    char_u *tail,                      /* tail of path */
-    int allow_dirs                         /* allow matching with dir */
-)
+/// Tries matching a filename with a "pattern" ("prog" is NULL), or use the
+/// precompiled regprog "prog" ("pattern" is NULL).  That avoids calling
+/// vim_regcomp() often.
+///
+/// Used for autocommands and 'wildignore'.
+///
+/// @param pattern    the pattern to match with
+/// @param prog       the pre-compiled regprog or NULL
+/// @param fname      the full path of the file name
+/// @param sfname     the short file name or NULL
+/// @param tail       the tail of the path
+/// @param allow_dirs allow matching with dir
+///
+/// @return true if there is a match, false otherwise
+static bool match_file_pat(char_u *pattern, regprog_T **prog, char_u *fname,
```

Comment:
A little gratuitous Doxygenification, change is `regprog_T *prog` -> `regprog_T **prog`